.github/workflows/build-libtorch-images.yml

```
@@ -0,0 +1,123 @@
+name: Build libtorch docker images
+
+on:
+  push:
+    branches:
+      - main
+      - release/*
+    tags:
+      # NOTE: Binary build pipelines should only get triggered on release candidate or nightly builds
+      # Release candidate tags look like: v1.11.0-rc1
+      - v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+
+    paths:
+      - '.ci/docker/libtorch/*'
+      - '.ci/docker/common/*'
+      - .github/workflows/build-libtorch-images.yml
+  pull_request:
+    paths:
+      - '.ci/docker/libtorch/*'
+      - '.ci/docker/common/*'
+      - .github/workflows/build-libtorch-images.yml
+
+env:
+  DOCKER_REGISTRY: "docker.io"
+  DOCKER_BUILDKIT: 1
+  DOCKER_ID: ${{ secrets.DOCKER_ID }}
+  DOCKER_TOKEN: ${{ secrets.DOCKER_TOKEN }}
+  WITH_PUSH: ${{ github.event_name == 'push' && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/release')) }}
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}-${{ github.event_name == 'workflow_dispatch' }}
+  cancel-in-progress: true
+
+jobs:
+  build-docker-cuda:
+    runs-on: linux.12xlarge
+    strategy:
+      matrix:
+        cuda_version: ["12.4", "12.1", "11.8"]
+    env:
+      GPU_ARCH_TYPE: cuda
+      GPU_ARCH_VERSION: ${{ matrix.cuda_version }}
+    steps:
+      - name: Checkout PyTorch
+        uses: pytorch/pytorch/.github/actions/checkout-pytorch@move_docker_conda
```

Why specify branch if it could be done from the same repo?
```suggestion
        uses: ./.github/actions/checkout-pytorch
```

---

Looks like this is what we use everywhere:
```
git grep "checkout-pytorch"
github/actions/teardown-win/action.yml:    # retry this step several time similar to how checkout-pytorch GHA does
.github/workflows/_android-build-test.yml:        uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
.github/workflows/_android-build-test.yml:        uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
.github/workflows/_android-full-build-test.yml:        uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
.github/workflows/_android-full-build-test.yml:        uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
.github/workflows/_bazel-build-test.yml:        uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
.github/workflows/_bazel-build-test.yml:        uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
.github/workflows/_binary-build-linux.yml:        uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
.github/workflows/_binary-test-linux.yml:        uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
.github/workflows/_binary-upload.yml:        uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
.github/workflows/_buck-build-test.yml:        uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
.github/workflows/_buck-build-test.yml:        uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
.github/workflows/_docs.yml:        uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
etc.....
```